### DISCUSSION TITLE
[INPUT NEEDED] A first pilot stock in your region that could use FIMS

### DISCUSSION THREAD

**Discussion by ChristineStawitz-NOAA** [NOAA-FIMS/FIMS#566] [General]
To facilitate M2 planning, we would like rep(s) from each region to answer the following questions by 7/26:

1. Which stock(s) in your region are a good first target to run FIMS alongside the tactical assessment?
2. What is missing from the milestone comparison project scope that will be completed in M1 that would need to be added to match the structure of the tactical model? Next to each feature please include whether this is a Must have (M), Should do (S), Could do (C), or Won't do (MoSCoW) in order to get an accurate bridge assessment.

**Comment by iantaylor-NOAA** [comment]
For NWFSC, Pacific hake is the only stock with a purely age-based assessment so seems like an obvious first choice.
The ~two~ three key missing features include:
- selectivity that is a flexible function of age and year (M)
- a Bayesian framework for specifying parameter priors (S) and sampling from the posterior distribution (W)
- [edit] cohort-specific ageing error matrices (C)

@kellijohnson-NOAA, let me know if I missed something here or if you have a different opinion

**Comment by EJDick-NOAA** [comment]
The SWFSC's assessment of Pacific Sardine is also uses only age comps (no lengths), and empirical weight at age. Regarding missing features, I think these are not currently in the plan for M1:

- fleet- and time-specific ageing error
- two seasons per year

There may be other minor differences, but those are what I could see after a quick skim.

https://www.fisheries.noaa.gov/resource/document/assessment-pacific-sardine-resource-2020-us-management-2020-2021

**Comment by nathanvaughan-NOAA** [comment]
For the SE Gulf we don't have any assessments that don't require at least length comps and a growth function (and probably a bunch of other post M1 features). @KyleShertzer-NOAA  may know of something in the South Atlantic that could be appropriate?

**Comment by jimianelli** [comment]
In the Bering Sea and Aleutian Islands LME we have an assessment for Atka mackerel that could be suitable. Fishery age comps and noisy survey observations (but with good ages) every 2 or 3 years. 

https://www.fisheries.noaa.gov/resource/data/2021-assessment-atka-mackerel-stock-bering-sea-and-aleutian-islands

for detail

**Comment by Cole-Monnahan-NOAA** [comment]
An alternative for the AFSC is the [Gulf of Alaska walleye pollock assessment](https://apps-afsc.fisheries.noaa.gov/refm/docs/2021/GOApollock.pdf). Basically no lengths, good ages. Empirical weight at age.

Would need time varying catchability and selectivity [S] and an ageing error matrix [S].

It would be fairly easy to walk the current model back to FIMS if that's something you want us to consider.

**Comment by cmlegault** [comment]
There are a number of stocks at the NEFSC that currently use ASAP and are well-suited for FIMS. These include Southern New England-Mid Atlantic yellowtail flounder, Atlantic herring, Gulf of Maine haddock, pollock, white hake, Gulf of Maine cod, black sea bass, scup, fluke, bluefish, Atlantic mackerel, and Southern New England-Mid Atlantic winter flounder. We will pick one lucky winner to be the official test case.

In terms of what is needed in milestone 2, the greatest need is for time-varying selectivity. ASAP uses time blocks to allow selectivity to change over time. The next highest need is the ability to estimate reference points and provide catch advice, typically with one year of catch (the year between the end of the assessment and the first year of new quota) then application of F40%SPR. 

@timjmiller do you see any other features needed in milestone 2 for FIMS to be able to replicate current ASAP runs?

**Comment by JonBrodziak** [comment]
For PIFSC the pilot stock is Main Hawaiian Islands Uku (Aprion virescens), a lutjanid last assessed in 2020. As for additional assessment features, it would be good to have capacity for length-based selectivity and multinomial size compositions (C).
Thanks and aloha,
Jon and Meg

**Comment by kellijohnson-NOAA** [comment]
@ChristineStawitz-NOAA what do you mean by "tactical model"? I agree with @iantaylor-NOAA that Pacific Hake is a good candidate but I do not necessarily think that we need Bayesian capability for M1, we can run the assessment in MLE mode easily, it just would lead to a different comparison than the model used for management.

**Comment by ChristineStawitz-NOAA** [comment]
Good question @kellijohnson-NOAA - I mean that we would present it to the councils as an appendix in 2023 and not use FIMS for tactical assessments until 2024. 

For the 2023 pilot, I think it is important to note as @Cole-Monnahan-NOAA mentions that we might need to simplify the existing models to get them to match FIMS. For example, if we don't have ageing error in FIMS in 2023, can we match FIMS to a version of the SS3 model without ageing error?

As another example, curious if you think it'd be worth considering petrale for west coast. I simplified it enough in our 2019 paper to be pretty confident removing ageing error and time-varying selectivity isn't going to have a huge impact on the biomass trajectory since the data are so good.

**Comment by iantaylor-NOAA** [comment]
@ChristineStawitz-NOAA: "can we match FIMS to a version of the SS3 model without ageing error" 
Yes, but I would worry that, at least for Pacific hake at least, the differences in model results from the base model would be dramatic enough that it could cause confusion to have those model results going before our management council. However, as @kellijohnson-NOAA points out, an MLE model comparison would be fine as those results are typically pretty comparable to the MCMC, so I'll downgrade "Bayesian framework" from "must do" to "won't do" in the comment above: https://github.com/NOAA-FIMS/FIMS/issues/203#issuecomment-1189558988.

I listed cohort-specific ageing error for Pacific hake as "could do". For sure we could live without the cohort-specific element, but I think that implementing a basic ageing error matrix in M2 would be really easy. In it's simplest form it could just be an input vector of standard deviation of the variability in observed age at each true age.

A challenge of using Petrale as a NWFSC pilot (and Main Hawaiian Islands Uku for a PIFSC pilot), is that they both use length-based processes which would require FIMS to have a growth module. At least the Uku model has growth parameters fixed (Table 1 in https://repository.library.noaa.gov/view/noaa/24190), so it would be theoretically possible to just input a matrix of the distribution of length at each age and not actually model growth parametrically.

**Comment by Cole-Monnahan-NOAA** [comment]
In my opinion I think it's perfectly acceptable for this phase to have a simplified version of a model that matches a FIMS version.  I suspect as we go to match them very closely we'll find hidden features that FIMS doesn't have and will need to be added as requirements.

I might have missed the memo but it seems unlikely we can use FIMS for tactical assessments in 2023 or 2024. It'll take a lot of bridging, testing, and explaining before the management bodies would accept that I think.

**Comment by JonBrodziak** [comment]
The pilot model exercise provides a proof-of-concept for FIMS in a real-world assessment setting. The fitted FIMS models are presumably being put forward to showcase the capacity of the nascent FIMS project.

As for Uku, having length-based selectivity and size composition fitting is a "could do". We can always apply a classical age-length key approach with a linear CV for errors about the fixed growth curve to convert the length compositions to age compositions, sans souci and fit the model in an age-based selectivity framework within FIMS

Aloha,
Jon

**Comment by Huihualee-NOAA** [comment]
North Pacific albacore and Pacific bluefin tunas in SWFSC are not purely age-based (estimate both size selectivity and age selectivity using quarterly size compositions (no age data)). I second EJ that Pacific sardine might be a start.

**Comment by KyleShertzer-NOAA** [comment]
A good candidate from the SE Atlantic region would be scamp grouper.  It will take some revision to match features of M1, but that's probably true of any real assessment.  I agree with Cole that simplified assessments are fine for now. At this stage, demonstrating that FIMS can match a simplified model would be far more useful than demonstrating that FIMS can't match a real assessment. 

For scamp, notable missing features include length comps and time-varying selectivity.

**Comment by cmlegault** [comment]
Atlantic mackerel will be the stock used for the NEFSC. It is currently in ASAP but will be transitioning to WHAM, so will allow comparisons among ASAP, WHAM, and FIMS.

**Comment by kellijohnson-NOAA** [comment]
Thank you everyone for your ideas and work on creating the first set of case studies for FIMS. I am going to close this issue now that all of the pilot case studies have been implemented. I have added a few issues to the case studies repository that highlight other stocks that could be added in the near future given comments here.
